src/firefly/java/edu/caltech/ipac/firefly/server/visualize/PlotServUtils.java

```diff
@@ -370,59 +213,51 @@ private static String getDateValueFromServiceFits(String headerKey, Header heade
         long year;
         String dateValue= header.getStringValue(headerKey);
         if(dateValue !=null){
-            if (headerKey.equals("ORDATE")) {
-                if (dateValue.length()>5) {
-                    dateValue= dateValue.subSequence(0,2)+"-"+dateValue.subSequence(2,4)+"-"+
-                            dateValue.subSequence(4,6);
-                    year = 2000+Integer.parseInt(dateValue.subSequence(0,2).toString());
-                    if (year > currentYear) {
-                        dateValue = "19"+dateValue;
-                    } else {
-                        dateValue = "20"+dateValue;
-                    }
-                }
-            } else if (headerKey.equals("DATE-OBS")) {
-                dateValue = dateValue.split("T")[0];
-                if (dateValue.contains("/")) {
-                    String newDate = "";
-                    for (String v: dateValue.split("/")) {
-                        if (newDate.length()==0) {
-                            newDate = v;
+            switch (headerKey) {
+                case "ORDATE":
+                    if (dateValue.length() > 5) {
+                        dateValue = dateValue.subSequence(0, 2) + "-" + dateValue.subSequence(2, 4) + "-" +
+                                dateValue.subSequence(4, 6);
+                        year = 2000 + Integer.parseInt(dateValue.subSequence(0, 2).toString());
+                        if (year > currentYear) {
+                            dateValue = "19" + dateValue;
                         } else {
-                            newDate = v + "-" + newDate;
+                            dateValue = "20" + dateValue;
                         }
                     }
-                    year = 2000+Integer.parseInt(newDate.subSequence(0,2).toString());
-                    if (year > currentYear) {
-                        dateValue = "19"+newDate;
-                    } else {
-                        dateValue = "20"+newDate;
+                    break;
+                case "DATE-OBS":
+                    dateValue = dateValue.split("T")[0];
+                    if (dateValue.contains("/")) {
+                        String newDate = "";
+                        for (String v : dateValue.split("/")) {
+                            if (newDate.length() == 0) {
+                                newDate = v;
+                            } else {
+                                newDate = v + "-" + newDate;
+                            }
+                        }
+                        year = 2000 + Integer.parseInt(newDate.subSequence(0, 2).toString());
+                        if (year > currentYear) {
+                            dateValue = "19" + newDate;
+                        } else {
+                            dateValue = "20" + newDate;
+                        }
```

Comment:
This is old code. I did not write it and I have looked at it very closely. I took so much out of `PlotServUtils.java` it probably looks new just because I rearrange a little.